frame/pablo/src/lib.rs

```diff
@@ -966,6 +966,39 @@ pub mod pallet {
 			Pools::<T>::contains_key(pool_id)
 		}
 
+		fn get_price_of_lp_token(pool_id: Self::PoolId) -> Result<Self::Balance, DispatchError> {
+			let pool = Self::get_pool(pool_id)?;
+			let pool_account = Self::account_id(&pool_id);
+			match pool {
+				PoolConfiguration::StableSwap(StableSwapPoolInfo { pair, lp_token, .. }) => {
+					let pool_base_aum =
+						T::Convert::convert(T::Assets::balance(pair.base, &pool_account));
+					let pool_quote_aum =
+						T::Convert::convert(T::Assets::balance(pair.quote, &pool_account));
+					let lp_total_issuance =
+						T::Convert::convert(T::Assets::total_issuance(lp_token));
+					let pool_general_aum = pool_base_aum.safe_add(&pool_quote_aum)?;
+					Ok(T::Convert::convert(pool_general_aum.safe_div(&lp_total_issuance)?))
```

Comment:
NIT:
Duplicate code can be moved to a new function

Reply:
Resolved